src/par_iter/internal.rs

```diff
@@ -132,7 +132,7 @@ pub trait UnindexedProducer: Send + Sized {
     type Item;
 
     /// Split midway into a new producer if possible, otherwise return `None`.
-    fn split(&mut self) -> Option<Self>;
+    fn split(self) -> (Self, Option<Self>);
```

Comment:
This `UnindexedProducer::split` interface change was because I found it awkward to deal with lifetimes in quicksort with extra `&mut` indirection, whereas it was much easier to deal with by value.